lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -61,21 +64,8 @@ sub list {
             for my $id (qw(machine_id test_suite_id product_id group_id)) {
                 if (my $value = $self->param($id)) { $cond{$id} = $value }
             }
-
-            my $has_query = grep { $cond{$_} } (
-                qw(machine_name machine_id test_suite.name test_suite_id group_id product.arch product.distri),
-                qw(product.flavor product.version product_id)
-            );
-
-            if ($has_query) {
-                my $attrs
-                  = {join => ['machine', 'test_suite', 'product'], prefetch => [qw(machine test_suite product)]};
-                @templates = $schema->resultset("JobTemplates")->search(\%cond, $attrs);
-            }
-            else {
-                @templates
-                  = $schema->resultset("JobTemplates")->search({}, {prefetch => [qw(machine test_suite product)]});
-            }
+            my %attrs = (prefetch => [qw(machine test_suite product)], rows => $limit);
+            @templates = $schema->resultset("JobTemplates")->search(\%cond, \%attrs);
```

Comment:
Like I said in the programming session, I double-checked the `$has_query` variable and can't see any conditions when there would be keys with empty values in the `%cond` hash, and if `%cond` itself is completely empty that should also not be a problem